telemetry_core/src/main/java/com/newrelic/telemetry/metrics/json/MetricToJson.java

```diff
@@ -1,100 +1,93 @@
+ *   Licensed under the Apache 2.0 License. See LICENSE in the project root directory for license information.
+ *  --------------------------------------------------------------------------------------------
  */
 
 package com.newrelic.telemetry.metrics.json;
 
+import com.newrelic.telemetry.json.AttributesJson;
+import com.newrelic.telemetry.json.JsonWriter;
 import com.newrelic.telemetry.metrics.Count;
 import com.newrelic.telemetry.metrics.Gauge;
 import com.newrelic.telemetry.metrics.Summary;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
+import java.io.IOException;
+import java.io.StringWriter;
 
-/**
- * If you wish to provide your own json implementation, an implementation of this interface must be
- * provided. All of the methods must return valid JSON with properly escaped strings.
- *
- * <p>TODO: add links to external documentation of the json format and APIs.
- */
-public interface MetricToJson {
-  Logger logger = LoggerFactory.getLogger(MetricToJson.class);
+/** This class turns Metrics into JSON via an embedded JsonWriter from the gson project. */
+public class MetricToJson {
+
+  private final AttributesJson attributeJson = new AttributesJson();
+
+  public String writeSummaryJson(Summary summary) {
+    try {
+      StringWriter out = new StringWriter();
+      JsonWriter jsonWriter = new JsonWriter(out);
+      jsonWriter.beginObject();
+      jsonWriter.name("name").value(summary.getName());
+      jsonWriter.name("type").value("summary");
+
+      jsonWriter.name("value");
+      jsonWriter.beginObject();
+      jsonWriter.name("count").value(summary.getCount());
+      jsonWriter.name("sum").value(summary.getSum());
+      jsonWriter.name("min").value(summary.getMin());
+      jsonWriter.name("max").value(summary.getMax());
+      jsonWriter.endObject();
+
+      jsonWriter.name("timestamp").value(summary.getStartTimeMs());
+      jsonWriter.name("interval.ms").value(summary.getEndTimeMs() - summary.getStartTimeMs());
+      String attributes = attributeJson.toJson(summary.getAttributes());
+      if (!attributes.isEmpty()) {
+        jsonWriter.name("attributes").jsonValue(attributes);
+      }
+      jsonWriter.endObject();
+      return out.toString();
+    } catch (IOException e) {
+      throw new RuntimeException("Failed to generate summary json", e);
+    }
+  }
 
-  /**
-   * Summary json must be structured this to be accepted by the New Relic backend.
-   *
-   * <p>the timestamp should be in ms since Unix epoch
-   *
-   * <p>The attributes are optional, if not present
-   *
-   * <pre>
-   * {
-   *   "name": "mySummaryName",
-   *   "type": "summary",
-   *   "value: {
-   *     "count": 33,
-   *     "sum": 1234,
-   *     "min": 10,
-   *     "max": 80
-   *   },
-   *   "timestamp": 1560807066910,
-   *   "interval.ms": 5000,
-   *   "attributes": {
-   *     "key1": "value1",
-   *     "key2", 555,
-   *     "key3", false
-   *   }
-   * }
-   * </pre>
-   */
```

Comment:
Same question about this javadoc -- shouldn't we keep it for helpful clarity?

Reply:
It would only be for us, since no one ever needs to write a json thingee any more (except us)